### PR TITLE
Arabic locale file

### PR DESCRIPTION
**Pull request by JeanNetryValere**
| Q               | A
| --------------- | ---
| Bug fix?        | [ ]
| New feature?    | [X ]
| New sample?      | [ ]
| Related issues?  | #461

#### What's in this Pull Request?

Adding Arabic translation support.

### CODE REVIEW

**Comment by JeanNetryValere** [comment]
@milanholemans Done here

**Comment by milanholemans** [comment]
That's better! ☺️
Don't forget to reference your issue #461 in the description of this PR. Then it should be all good!

**Comment by joelfmrodrigues** [comment]
@JeanNetryValere many thanks for the PR. Great addition.
Unfortunately, I don't know a single word of Arabic 🙂 so I have reached out for help on Twitter to see if someone is available to review as it's always good to have someone reviewing changes. If no one volunteers to help, I will just approve it 🙂

**Comment by joelfmrodrigues** [comment]
@milanholemans many many thanks for your support on this issue and PR. We have been all super busy with work and having other people stepping in to help is fantastic!
GoodJobYouRockGIF

File: src/loc/ar-sa.js [hunk 1]
```diff
@@ -0,0 +1,246 @@
+define([],function () {
+  return {
+    ApplyButtonLabel: "طبق",
+    ImportButtonLabel: "حمل",
+    ExportButtonLabel: "صدر",
+    FormatCodeButtonLabel: "كود التنسيق",
+    JsonFileRequiredMessage: "الرجاء تحميل ملف json",
+    EditPropertiesButtonLabel: "تحرير الخصائص",
+    EditPropertiesPanelHeaderText: "تحرير الخصائص الأولية",
+    SaveButtonLabel: "يحفظ",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:10]: "حفظ",

File: src/loc/ar-sa.js [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+  return {
+    ApplyButtonLabel: "طبق",
+    ImportButtonLabel: "حمل",
+    ExportButtonLabel: "صدر",
+    FormatCodeButtonLabel: "كود التنسيق",
+    JsonFileRequiredMessage: "الرجاء تحميل ملف json",
+    EditPropertiesButtonLabel: "تحرير الخصائص",
+    EditPropertiesPanelHeaderText: "تحرير الخصائص الأولية",
+    SaveButtonLabel: "يحفظ",
+    CancelButtonLabel: "يلغي",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:11]: "إلغاء",

File: src/loc/ar-sa.js [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    DatePickerMonthLongFebruary: "فبراير",
+    DatePickerMonthShortFebruary: "فبراير",
+    DatePickerMonthLongMarch: "مارس",
+    DatePickerMonthShortMarch: "مارس",
+    DatePickerMonthLongApril: "أبريل",
+    DatePickerMonthShortApril: "أبريل",
+    DatePickerMonthLongMay: "مايو",
+    DatePickerMonthShortMay: "مايو",
+    DatePickerMonthLongJune: "يونيه",
+    DatePickerMonthShortJune: "يونيو",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:28]: "يونيه",

File: src/loc/ar-sa.js [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+    DatePickerMonthLongJuly: "يوليو",
+    DatePickerMonthShortJuly: "يوليو",
+    DatePickerMonthLongAugust: "أغسطس",
+    DatePickerMonthShortAugust: "أغسطس",
+    DatePickerMonthLongSeptember: "سبتمبر",
+    DatePickerMonthShortSeptember: "سبتمبر",
+    DatePickerMonthLongOctober: "اكتوبر",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:35]: "أكتوبر",

File: src/loc/ar-sa.js [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+    DatePickerMonthShortOctober: "أكتوبر",
+    DatePickerMonthLongNovember: "نوفمبر",
+    DatePickerMonthShortNovember: "نوفمبر",
+    DatePickerMonthLongDecember: "ديسمبر",
+    DatePickerMonthShortDecember: "ديسمبر",
+    DatePickerDayLongSunday: "الأحد",
+    DatePickerDayShortSunday: "الأحد",
+    DatePickerDayLongMonday: "الاثنين",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:43]: "الأثنين",

File: src/loc/ar-sa.js [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+    DatePickerDayShortMonday: "الاثنين",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:44]: "الأثنين",

File: src/loc/ar-sa.js [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+    DatePickerDayLongTuesday: "الثلاثاء",
+    DatePickerDayShortTuesday: "الثلاثاء",
+    DatePickerDayLongWednesday: "الأربعاء",
+    DatePickerDayShortWednesday: "الأربعاء",
+    DatePickerDayLongThursday: "الخميس",
+    DatePickerDayShortThursday: "الخميس",
+    DatePickerDayLongFriday: "جمعة",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:51]: "الجمعة",

File: src/loc/ar-sa.js [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+    DatePickerDayShortFriday: "الجمعة",
+    DatePickerDayLongSaturday: "السبت",
+    DatePickerDayShortSaturday: "السبت",
+    DatepickerGoToToday: "اليوم",
+    DateTimePickerDate: "تاريخ",
+    DateTimePickerTime: "زمن",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:57]: "الوقت",

File: src/loc/ar-sa.js [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+    SitePickerNoResults: "لا نتائج للبحث",
+    SitePickerSitesChosen: "تم تحديد موقع (مواقع) الويب",
+    TeamPickerSearchBoxPlaceholder: "يبحث...",
+    TeamPickerNoResults: "لا نتائج للبحث",
+    TeamPickerSitesChosen: "تم تحديد الفرق",
+
+    AddFileButtonLabel: "اضف ملف",
+    AddImageButtonLabel: "إضافة صورة",
+    AriaCellValue: "{0} العمود ، {1}",
+    FilePickerCancelButtonLabel: "يلغي",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:92]: "إلغاء",

File: src/loc/ar-sa.js [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+    FolderIconUrl: "https://spoprod-a.akamaihd.net/files/odsp-next-prod_2019-01-11_20190116.001/odsp-media/images/itemtypes/20/folder.svg",
+    FolderLabelTemplate: "{0} ، مجلد ، معدل {1} ، تم التعديل بواسطة {2} ، {3} عنصر ، خاص",
+    FromLinkLinkLabel: "من رابط",
+    ImageAltText: ".{0} صورة",
+    ImageAriaLabelTemplate: ".{0} صورة",
+    ImageLayoutPlaceholderText: "تَخطِيط",
+    ImageSizePlaceholderText: "حجم الصورة",
+    ItemChildCountField: "عدد العناصر التابعة",
+    LayoutOptionAll: "الجميع",
+    LayoutOptionSquare: "ميدان",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:120]: "مربع",

File: src/loc/ar-sa.js [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+    LayoutOptionTall: "طويل",
+    LayoutOptionWide: "واسع",
+    LearnMoreLink: "يتعلم أكثر.",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:123]: "تعلم المزيد.",

File: src/loc/ar-sa.js [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+    ListLayoutCompactDescription: "عرض العناصر والتفاصيل في قائمة مضغوطة",
+    ListLayoutList: "عرض القائمة",
+    ListLayoutListDescrition: "عرض العناصر والتفاصيل في قائمة",
+    ListLayoutTile: "عرض التجانب",
+    ListLayoutTileDescription: "عرض العناصر بمعاينات البلاط",
+    ListOptionsAlt: "خيارات العرض. {0} محددة.",
+    ListOptionsTitle: "فتح قائمة خيارات العرض",
+    Loading: "جار التحميل...",
+    ModifiedByField: "تم التعديل بواسطة",
+    ModifiedField: "التاريخ عدل",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:142]: "التاريخ التعديل",

File: src/loc/ar-sa.js [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+    No: "رقم",
+    ProvidedValueIsInvalid: "القيمة المقدمة غير صالحة",
+    NoExternalLinksValidationMessage: "نحن ندعم فقط الارتباط بالملفات الموجودة في مؤسستك.",
+    NoImageValidationMessage: "هذا ليس رابطًا لنوع ملف ندعمه. يمكنك فقط الارتباط بصورة.",
+    NoRecentFiles: "لا توجد ملفات حديثة",
+    NoRecentFilesDescription: "حاول تحديد ملف من موقعك ، أو تحميل واحد من جهازك.",
+    NoResultsBadEnglish: "*** لم يتم العثور على نتيجة. حاول تغيير خيارات التصفية",
+    ODModifiedField: "المعدل",
+    ODPhotoIconUrl: "https://spoprod-a.akamaihd.net/files/odsp-next-prod_2019-01-18_20190124.001/odsp-media/images/itemtypes/16_2x/photo.png",
+    ODRowArialLabelTemplate: "{0} ، {1} ، تم التعديل {2} ، تم التعديل بواسطة {3} ، {4} ، {5}",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:153]: "{0} ، {1} ، تم التعديل {2} ، تم التعديل بواسطة {3} ، {4} ، {5}" ،

File: src/loc/ar-sa.js [hunk 14]
```diff
@@ -0,0 +1,246 @@
...
+    OpenButtonLabel: "فتح",
+    PhotoIconUrl: "https://spoprod-a.akamaihd.net/files/odsp-next-prod_2019-01-11_20190116.001/odsp-media/images/itemtypes/20_2x/photo.png",
+    OrgAssetsTabLabel: "الصور والملفات المقدمة من مؤسستك",
+    PoweredByBing: "بدعم من Bing",
+    RecentDocumentsHeader: "الوثائق الأخيرة",
+    RecentImagesHeader: "الصور الحديثة",
+    RecentLinkLabel: "مؤخرًا",
+    SearchBoxPlaceholder: "البحث في الويب",
+    SearchResultAlt: "نتيجة الصورة لـ {0}.",
+    SearchResultAriaLabel: "اضغط على Enter لفتح مصدر الصورة في علامة تبويب جديدة.",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:170]: "اضغط على زر الإدخال لفتح مصدر الصورة في علامة تبويب جديدة."

File: src/loc/ar-sa.js [hunk 15]
```diff
@@ -0,0 +1,246 @@
...
+      "L_RelativeDateTime_XDaysIntervals": "1||2-",
+      "L_RelativeDateTime_Today": "اليوم"
+    },
+
+    NewFolderNamePlaceholder: "أدخل اسم المجلد الخاص بك",
+    FolderFilterBoxPlaceholder: "تصفية المجلدات حسب الاسم",
+    FolderExplorerLoading: "تحميل المجلدات ...",
+    FolderExplorerNoItems: "لا يحتوي هذا المجلد على أي مجلدات فرعية.",
+
+    IncorrectGuidValidationMessage: "الرجاء إدخال GUID صالح",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:236]: "الرجاء إدخال معرف صالح" ،

File: src/loc/ar-sa.js [hunk 16]
```diff
@@ -0,0 +1,246 @@
...
+
+    SelectedLabel: "المحدد",
+    SelectIcon: "تحديد الرمز",
+    CloseButton: "قريب",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:240]: "إغلاق",

File: src/loc/ar-sa.js [hunk 17]
```diff
@@ -0,0 +1,246 @@
...
+    "MonacoEditorOpenButtonLabel": "فتح المحرر",
+    "MonacoEditorCancelButtonLabel": "يلغي",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:242]: "إلغاء",

File: src/loc/ar-sa.js [hunk 18]
```diff
@@ -0,0 +1,246 @@
...
+    "MonacoEditorSaveButtonLabel": "يحفظ",
```
Comment (mohammadamer) [inline src/loc/ar-sa.js:243]: "حفظ",

**Review by mohammadamer — COMMENTED** [review]
Locale has been reviewed

**Comment by JeanNetryValere** [comment]
Hello @joelfmrodrigues,
Yes perfect, it has been reviewed by my team also.
Thank you @mohammadamer

**Review by JeanNetryValere — COMMENTED** [review]
Changes accepted

**Comment by JeanNetryValere** [comment]
@joelfmrodrigues review submitted
Please let me now if closing PR is required

**Comment by milanholemans** [comment]
Hi @JeanNetryValere 
Sorry that I'm making this comment so late, but commenting "changes accepted" is unfortunately not enough. 😃
What you should do now is review all changes of Mohammad and make these changes in your code (be sure to target the same branch as this PR so your `master` branch).
When you made these adjustments in your code, simply do a new commit to this branch. The PR will update and we'll be able to see the new code.
When the new code is fine, the maintainers of this repo will accept the PR and close it, so you shouldn't close the PR yourself.

If you have any questions, just ask.

**Comment by joelfmrodrigues** [comment]
@milanholemans many thanks for the info provided.
@JeanNetryValere does the info provided from Milan make sense? Do you have any questions or need assistance with something?

**Comment by AJIXuMuK** [comment]
Hello @JeanNetryValere - following up on this one.
Have you made the proposed changes?

Thanks!

**Comment by JeanNetryValere** [comment]
> Hello @JeanNetryValere - following up on this one. Have you made the proposed changes?
> 
> Thanks!

I did, please revert to me if required.
Have fun on merge :)

**Comment by AJIXuMuK** [comment]
Thank you @JeanNetryValere for the confirmation!

**Pull request merged**